packages/taquito/test/prepare/prepare-provider.spec.ts

```diff
@@ -0,0 +1,1000 @@
+              fee: '391',
+              gas_limit: '101',
+              storage_limit: '1000',
+              delegate: 'KT1Fe71jyjrxFg9ZrYqtvaX7uQjcLo7svE4D',
+              counter: '1',
+            },
+          ],
+          protocol: 'test_protocol',
+        },
+        counter: 0,
+      });
+    });
+  });
+
+  describe('registerGlobalConstant', () => {
+    it('should return a prepared registerGlobalConstant operation', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(estimate);
+
+      const prepared = await prepareProvider.registerGlobalConstant({
+        value: { prim: 'Pair', args: [{ int: '999' }, { int: '999' }] },
+      });
+
+      expect(prepared).toEqual({
+        opOb: {
+          branch: 'test_block_hash',
+          contents: [
+            {
+              kind: 'reveal',
+              fee: '391',
+              public_key: 'test_pub_key',
+              source: 'test_public_key_hash',
+              gas_limit: '101',
+              storage_limit: '1000',
+              counter: '1',
+            },
+            {
+              kind: 'register_global_constant',
+              value: {
+                prim: 'Pair',
+                args: [
+                  {
+                    int: '999',
+                  },
+                  {
+                    int: '999',
+                  },
+                ],
+              },
+              fee: '391',
+              gas_limit: '101',
+              storage_limit: '1000',
+              source: 'test_public_key_hash',
+              counter: '2',
+            },
+          ],
+          protocol: 'test_protocol',
+        },
+        counter: 0,
+      });
+    });
+
+    it('should be able to prepare register_global_constant op ', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(undefined);
+
+      const prepared = await prepareProvider.registerGlobalConstant({
+        value: { prim: 'Pair', args: [{ int: '999' }, { int: '999' }] },
+      });
+
+      expect(prepared).toEqual({
+        opOb: {
+          branch: 'test_block_hash',
+          contents: [
+            {
+              kind: 'register_global_constant',
+              value: {
+                prim: 'Pair',
+                args: [
+                  {
+                    int: '999',
+                  },
+                  {
+                    int: '999',
+                  },
+                ],
+              },
+              fee: '391',
+              gas_limit: '101',
+              storage_limit: '1000',
+              source: 'test_public_key_hash',
+              counter: '1',
+            },
+          ],
+          protocol: 'test_protocol',
+        },
+        counter: 0,
+      });
+    });
+  });
+  describe('updateConsensusKey', () => {
+    it('should return a prepared udpateConsensusKey operation', async () => {
+      jest.spyOn(context.estimate, 'reveal').mockResolvedValue(estimate);
```

Comment:
Not sure update_consensus key op need to reveal either, since it's the delegate initiating this op it has been revealed.

Reply:
update_consensus_key should have estimates. So I think this is still relevant. But you are correct with drain_delegate